### PR TITLE
feat: add open map button #279

### LINKED ISSUE #279: Redirect user to open map provider based on address
## Description

image

## Implementation Model

User might want to know address location from map provider such google maps.

## Tasks

- [ ] redirect user to map provider with text address as param like `https://www.google.com/maps/place/${address}`

### PR DESCRIPTION
**Pull request by fncolon**
Closes #279 

cc @andriawan  @resir014 

## Description

Add Open Map Button Based On Address

## Current Tasks

- [x] redirect user to map provider with text address as param like https://www.google.com/maps/place/${address}

Song (avoid copyright) : SO-SO - This is 8-bit

https://user-images.githubusercontent.com/50759463/126634486-7b1cbbdd-5d3c-449b-920b-b63448bd6bc6.mp4

### CODE REVIEW

File: components/open-map-button.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { SecondaryButton } from "./ui/button";
+
+import { MapIcon } from "@heroicons/react/outline";
+
+export function OpenMapButton({ address }: { address: string }) {
+  function btnClick(e: React.FormEvent<HTMLButtonElement>) {
+    e.currentTarget.blur();
+    window.open(`https://www.google.com/maps/place/${address}`);
```
Comment (ekamuktia) [inline components/open-map-button.tsx:8]: Please use `https://www.google.com/maps/search` instead of `https://www.google.com/maps/place`

**Review by ekamuktia — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/9606523/126653048-5c25b74d-fe03-482c-9172-e984340aa3c7.mp4

As can be seen on video, if the place exists on google map current url will be fine. But if the place doesn't exist, it doesn't show properly.

Please change the link to google map. More advanced parameters can be checked here https://developers.google.com/maps/documentation/urls/get-started

File: components/open-map-button.tsx (hunk 1 above)
Comment (fncolon) [reply components/open-map-button.tsx]: @ekamuktia resolved, thanks

File: components/open-map-button.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    <div className="flex items-start">
+      <SecondaryButton
+        aria-label="Buka Map"
+        className="relative z-10"
+        icon={MapIcon}
+        onClick={btnClick}
+        size="xs"
+        type="button"
+      >
+        Buka Map
```
Comment (zainfathoni) [inline components/open-map-button.tsx:21]:
```suggestion
        Buka Peta
```

To make it consistent for the users, let's use Bahasa Indonesia words 😁

**Review by zainfathoni — CHANGES_REQUESTED** [review]

File: components/contact-details.tsx [hunk 3]
```diff
@@ -60,7 +67,12 @@ export function ContactDetails({ contact }: ContactDetailsProps) {
         <DescriptionItem label="Keterangan" value={contact.keterangan} />
         <DescriptionItem label="Lokasi" value={contact.lokasi} />
         <DescriptionItem label="Kontak" value={contact.kontak} withCopyButton />
-        <DescriptionItem label="Alamat" value={contact.alamat} withCopyButton />
+        <DescriptionItem
```
Comment (zainfathoni) [inline components/contact-details.tsx:70]: For the first iteration, I think it's still acceptable.
However, we should make it more consistent by putting it under the `Salin` button, so all action buttons are located on the right side.

@andriawan could you please create another issue to follow up on this?
Thanks!

Comment (zainfathoni) [reply components/contact-details.tsx:70]: Issue created on #344

File: components/open-map-button.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+export function OpenMapButton({ address }: { address: string }) {
+  function btnClick(e: React.FormEvent<HTMLButtonElement>) {
+    e.currentTarget.blur();
+    window.open(`https://www.google.com/maps/search/?api=1&query=${address}`);
+  }
+
+  return (
+    <div className="flex items-start">
+      <SecondaryButton
+        aria-label="Buka Map"
```
Comment (ekamuktia) [inline components/open-map-button.tsx:14]:
```suggestion
        aria-label="Buka Peta"
```
Please make it consistence with the button label

**Review by ekamuktia — CHANGES_REQUESTED** [review]

**Review by ekamuktia — APPROVED** [review]
LGTM

**Review by zainfathoni — APPROVED** [review]
LGTM 💯 

Thanks, @fncolon, for the code contribution!

@all-contributors, please add @fncolon for code.

**Comment by zainfathoni** [comment]
all-contributors, please add @fncolon for code.

**Comment by zainfathoni** [comment]
@all-contributors, please add @fncolon for code.

**Comment by zainfathoni** [comment]
@all-contributors, please add @fncolon for code.

**Comment by zainfathoni** [comment]
@all-contributors please add @fncolon for doc and code.

**Pull request merged**